lib/puppet/parser/functions/getvar_emptystring.rb

```diff
@@ -0,0 +1,35 @@
+module Puppet::Parser::Functions
+
+  newfunction(:getvar_emptystring, :type => :rvalue, :doc => <<-'ENDHEREDOC') do |args|
+    Lookup a variable in a remote namespace.
+    Return an empty string rather than undef/nil if it could not be found
+
+    For example:
+
+        $foo = getvar('site::data::foo')
+        # Equivalent to $foo = $site::data::foo
+
+    This is useful if the namespace itself is stored in a string:
+
+        $datalocation = 'site::data'
+        $bar = getvar("${datalocation}::bar")
+        # Equivalent to $bar = $site::data::bar
+    ENDHEREDOC
+
+    unless args.length == 1
+      raise Puppet::ParseError, ("getvar_emptystring(): wrong number of arguments (#{args.length}; must be 1)")
+    end
+
+    begin
+      result = self.lookupvar("#{args[0]}")
```

Comment:
This turns whatever args[0] is into a string. The function should check that the argument is a string of length >= 1. If also supporting numeric variables (match vars) it could allow integer values.